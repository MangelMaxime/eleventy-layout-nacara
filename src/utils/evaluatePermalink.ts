
export const evaluatePermalink = (ctx: any) => {
    let permalink = null;
    if (ctx.data.permalink) {
        if (typeof ctx.data.permalink === "function") {
            permalink = ctx.data.permalink(ctx.data)
        } else {
            permalink = ctx.data.permalink;
        }
    }
    return permalink;
}
