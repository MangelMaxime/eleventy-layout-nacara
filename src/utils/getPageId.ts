import path from "path";
// import NacaraTypes from "./../../types/nacara";

export default function getPageId(fileStem : string) {
    //  Normal the path, so we can split using the path separator
    const normalizedInputPath = path.normalize(fileStem);
    // Extract all the segments of the path
    const inputPathSegments = normalizedInputPath.split(path.sep);
    // console.log("inputPathSegments:", inputPathSegments);
    // Build the section direction, which consist of the root + the first segment of the path
    const pageIdSegments = inputPathSegments.slice(2);

    return pageIdSegments.join("/");
};
