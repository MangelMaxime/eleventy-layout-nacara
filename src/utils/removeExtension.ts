import path from "path";

export function removeExtension(fileName: string): string {
    const parsedPath = path.parse(fileName);
    return path.join(parsedPath.dir, parsedPath.name);
}
