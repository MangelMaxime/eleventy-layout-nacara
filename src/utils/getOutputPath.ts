import { evaluatePermalink } from "./evaluatePermalink.js";

export default function getOutputPath(pageContext: any) {
    const trimOutputRegex = new RegExp(`^${pageContext.data.eleventy.directories.output}`);
    const permalink = evaluatePermalink(pageContext);

    if (permalink) {
        return permalink;
    }

    return pageContext.data.page.outputPath.replace(trimOutputRegex, "");
}
