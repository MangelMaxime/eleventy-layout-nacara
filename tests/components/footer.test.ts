import { expect, test } from 'vitest'
import { formatHTML } from "../utils/_formatHTML";
const Eleventy = require("@11ty/eleventy");

test("render nothing if footer.nacara is not provided", async () => {
    const elev = new Eleventy(
        "./fixtures/footer-0/",
        "./fixtures/footer-0/_site",
        {
            configPath: "./fixtures/footer-0/.eleventy.js",
        }
    );

    const json = await elev.toJSON();
    const formattedResult = formatHTML(json[0].content);

    expect(formattedResult).toMatchSnapshot()
});

test("is rendered if only the text is provided", async () => {
    const elev = new Eleventy(
        "./fixtures/footer-1/",
        "./fixtures/footer-1/_site",
        {
            configPath: "./fixtures/footer-1/.eleventy.js",
        }
    );

    const json = await elev.toJSON();
    const formattedResult = formatHTML(json[0].content);

    expect(formattedResult).toMatchSnapshot()
});

test("is rendered if only copyright is provided", async () => {
    const elev = new Eleventy(
        "./fixtures/footer-2/",
        "./fixtures/footer-2/_site",
        {
            configPath: "./fixtures/footer-2/.eleventy.js",
        }
    );

    const json = await elev.toJSON();
    const formattedResult = formatHTML(json[0].content);

    expect(formattedResult).toMatchSnapshot();
});

test("is rendered if only the sitemapSections is provided", async () => {
    const elev = new Eleventy(
        "./fixtures/footer-3/",
        "./fixtures/footer-3/_site",
        {
            configPath: "./fixtures/footer-3/.eleventy.js",
        }
    );

    const json = await elev.toJSON();
    const formattedResult = formatHTML(json[0].content);

    expect(formattedResult).toMatchSnapshot();
});

test("is rendered with all the information provided", async () => {
    const elev = new Eleventy(
        "./fixtures/footer-4/",
        "./fixtures/footer-4/_site",
        {
            configPath: "./fixtures/footer-4/.eleventy.js",
        }
    );

    const json = await elev.toJSON();
    const formattedResult = formatHTML(json[0].content);

    expect(formattedResult).toMatchSnapshot();
});
