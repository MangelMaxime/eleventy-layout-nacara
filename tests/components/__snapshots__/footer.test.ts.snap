// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`is rendered if only copyright is provided 1`] = `
"<footer class="footer">
    <div class="is-size-5">
        <p class="has-text-centered">
            Copyright © 2023-
            <span id="copyright-end-year"></span>
            Mangel Maxime
        </p>

        <script type="text/javascript">
            // Use IIFE to avoid polluting the global scope
            // This also avoid conflict because of \`const year\` on the components/footer page
            (function () {
                const year = new Date().getFullYear();
                document.getElementById("copyright-end-year").innerHTML = year;
            })();
        </script>
    </div>
</footer>
"
`;

exports[`is rendered if only the sitemapSections is provided 1`] = `
"<footer class="footer">
    <div class="is-size-5">
        <div class="sitemap">
            <div class="sitemap-section">
                <div class="sitemap-section-title">About</div>
                <ul class="sitemap-section-list">
                    <li>
                        <a
                            href="/about-us"
                            class="icon-text sitemap-section-list-item"
                        >
                            <span class="sitemap-section-list-item-text">
                                About Us
                            </span>
                        </a>
                    </li>

                    <li>
                        <a
                            href="/our-people"
                            class="icon-text sitemap-section-list-item"
                        >
                            <span class="icon">
                                <svg
                                    class="lucide lucide-users"
                                    xmlns="http://www.w3.org/2000/svg"
                                    width="24"
                                    height="24"
                                    viewBox="0 0 24 24"
                                    fill="none"
                                    stroke="currentColor"
                                    stroke-width="2"
                                    stroke-linecap="round"
                                    stroke-linejoin="round"
                                >
                                    <path
                                        d="M16 21v-2a4 4 0 0 0-4-4H6a4 4 0 0 0-4 4v2"
                                    ></path>
                                    <circle cx="9" cy="7" r="4"></circle>
                                    <path d="M22 21v-2a4 4 0 0 0-3-3.87"></path>
                                    <path d="M16 3.13a4 4 0 0 1 0 7.75"></path>
                                </svg>
                            </span>

                            <span class="sitemap-section-list-item-text">
                                Our People
                            </span>
                        </a>
                    </li>
                </ul>
            </div>

            <div class="sitemap-section">
                <div class="sitemap-section-title">Ressources</div>
                <ul class="sitemap-section-list">
                    <li>
                        <a
                            href="/blog"
                            class="icon-text sitemap-section-list-item"
                        >
                            <span class="sitemap-section-list-item-text">
                                Blog
                            </span>
                        </a>
                    </li>

                    <li>
                        <a
                            href="/faq"
                            class="icon-text sitemap-section-list-item"
                        >
                            <span class="sitemap-section-list-item-text">
                                FAQ
                            </span>
                        </a>
                    </li>
                </ul>
            </div>
        </div>
    </div>
</footer>
"
`;

exports[`is rendered if only the text is provided 1`] = `
"<footer class="footer">
    <div class="is-size-5">
        <p class="has-text-centered">
            Built with
            <a
                class="is-underlined"
                href="https://mangelmaxime.github.io/Nacara/"
            >
                eleventy-layout-nacara
            </a>
        </p>
    </div>
</footer>
"
`;

exports[`is rendered with all the information provided 1`] = `
"<footer class="footer">
    <div class="is-size-5">
        <div class="sitemap">
            <div class="sitemap-section">
                <div class="sitemap-section-title">About</div>
                <ul class="sitemap-section-list">
                    <li>
                        <a
                            href="/about-us"
                            class="icon-text sitemap-section-list-item"
                        >
                            <span class="sitemap-section-list-item-text">
                                About Us
                            </span>
                        </a>
                    </li>

                    <li>
                        <a
                            href="/our-people"
                            class="icon-text sitemap-section-list-item"
                        >
                            <span class="icon">
                                <svg
                                    class="lucide lucide-users"
                                    xmlns="http://www.w3.org/2000/svg"
                                    width="24"
                                    height="24"
                                    viewBox="0 0 24 24"
                                    fill="none"
                                    stroke="currentColor"
                                    stroke-width="2"
                                    stroke-linecap="round"
                                    stroke-linejoin="round"
                                >
                                    <path
                                        d="M16 21v-2a4 4 0 0 0-4-4H6a4 4 0 0 0-4 4v2"
                                    ></path>
                                    <circle cx="9" cy="7" r="4"></circle>
                                    <path d="M22 21v-2a4 4 0 0 0-3-3.87"></path>
                                    <path d="M16 3.13a4 4 0 0 1 0 7.75"></path>
                                </svg>
                            </span>

                            <span class="sitemap-section-list-item-text">
                                Our People
                            </span>
                        </a>
                    </li>
                </ul>
            </div>
        </div>

        <p class="has-text-centered">Build with eleventy-layout-nacara</p>

        <p class="has-text-centered">
            Copyright © 2023-
            <span id="copyright-end-year"></span>
            Mangel Maxime
        </p>

        <script type="text/javascript">
            // Use IIFE to avoid polluting the global scope
            // This also avoid conflict because of \`const year\` on the components/footer page
            (function () {
                const year = new Date().getFullYear();
                document.getElementById("copyright-end-year").innerHTML = year;
            })();
        </script>
    </div>
</footer>
"
`;

exports[`render nothing if footer.nacara is not provided 1`] = `""`;
